docs/proposals/multi-platform.md

```diff
@@ -0,0 +1,554 @@
+
+Also, [the apiserver now exposes](https://github.com/kubernetes/kubernetes/pull/19905) it's platform at `GET /version`. But note that the value exposed at `/version` only is the apiserver's platform; there might be kubelets of various other platforms.
+
+### Set up a build and e2e CI's
+
+#### Build CI
+
+Kubernetes should always enforce that all binaries are compiling; otherwise action should be taken.
+The job that runs `make release` should be Submit Queue-blocking, and a PR that merges that only work on `linux/amd64` should be reverted in order to unblock the Submit Queue.
+
+For more information, see [conflicts](#conflicts)
+
+#### e2e CI
+
+To ensure all functionality really is working on all other platforms, the community should be able to setup a CI.
+To be able to do that, all the test-specific images have to be ported to multiple architectures, and the test images should preferably be manifest lists.
+If the test images aren't manifest lists, the test code should automatically choose the right image based on the image naming.
+
+IBM volunteered to run continuously running e2e tests for `linux/ppc64le`.
+Still it's hard to set up a such CI (even on `linux/amd64`), but that work belongs to `kubernetes/test-infra` proposals.
+
+When it's working to test Kubernetes using Kubernetes; volunteers should be given access to publish their results on `k8s-testgrid.appspot.com`.
+
+### Standardize all image Makefiles to follow the same pattern
+
+All Makefiles should push for all platforms when doing `make push`, and build for all platforms when doing `make build`.
+Under the hood; they should compile binaries in a container for reproducability, and use QEMU for emulating Dockerfile `RUN` commands if necessary.
+
+### Remove linux/amd64 hard-codings from the codebase
+
+All places where `linux/amd64` is hardcoded in the codebase should be rewritten.
+
+#### Make kubelet automatically use the right pause image
+
+The `pause` is used for connecting containers into Pods. It's a binary that just sleeps forever.
+When Kubernetes starts up a Pod, it first starts a `pause` container, and let's all "real" containers join the same network by setting `--net=${pause_container_id}`.
+
+So in order to start Kubernetes Pods on any other architecture, an ever-sleeping image have to exist.
+
+Fortunately, `kubelet` has the `--pod-infra-container-image` option, and it has been used when running Kubernetes on other platforms.
+
+But relying on the deployment setup to specify the right image for the platform isn't great, the kubelet should be smarter than that.
+
+This specific problem has been fixed in [#23059](https://github.com/kubernetes/kubernetes/pull/23059).
+
+#### Vendored packages
+
+Kubernetes might depend on packages that aren't coded in a cross-platform manner.
```

Comment:
No and yes.

They should be, but `make release` isn't run as a presubmit test right now.
Example PR were this happened: https://github.com/kubernetes/kubernetes/pull/29349

That turned out to be a really bad idea [when releasing](https://github.com/kubernetes/test-infra/issues/393#issuecomment-240320412) however, as it delayed the `v1.4.0-alpha.3` release with more than a week,

But I might add that the goal should be that the CI should directly catch such failures (although it isn't so today)